src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -19,10 +19,11 @@ public class CompilerDataLogger
         private const string CompilerEventName = "CompilerInformation";
 
         private readonly bool appInsightsRegistered;
-
         private readonly string sha256;
         private readonly string relativeFilePath;
 
+        private static string s_sessionId;
+        private static bool s_printHeader = true;
```

Comment:
>s_printHeader [](http://example.com/codeflow?start=28&length=13)

this will handle if we need to print the header.
Once we print, we will set to not print